htdocs/core/class/commondocgenerator.class.php

```diff
@@ -550,11 +550,11 @@ public function get_substitutionarray_object($object, $outputlangs, $array_key =
 			$array_key.'_total_localtax2_locale'=>price($object->total_localtax2, 0, $outputlangs),
 			$array_key.'_total_ttc_locale'=>price($object->total_ttc, 0, $outputlangs),
 
-			$array_key.'_total_ht'=>price2num($object->total_ht),
-			$array_key.'_total_vat'=>(!empty($object->total_vat) ?price2num($object->total_vat) : price2num($object->total_tva)),
+			$array_key.'_total_ht'=>price($object->total_ht, 0, $outputlangs),
```

Comment:
The var total_ht is an amount. It lust be the numeric value, so not a formated value that is a string.
For a formated value, with local language, you should use
total_ht_locale